### ISSUE TITLE
comment-spacings false positives for nolint directives

### ISSUE THREAD

**Issue by abemedia** [mgechev/revive#808]
**Describe the bug**
I'm using revive with golangci-lint and since the latest update I'm getting errors like this:
```
api/common/validate/validate.go:15:1: comment-spacings: no space between comment delimiter and comment text (revive)
//nolint:gochecknoglobals
```
I've updated the config to contain `nolint` in the `comment-spacings` arguments to make it go away but it's odd as this was not happening before.

**To Reproduce**
1. I updated golangci-lint:
```sh
curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin
```

2. I run it with the following flags & configuration file:
```shell
golangci-lint run
```

```yaml
linters:
  disable-all: true
  enable:
    - revive
linters-settings:
  revive:
    rules:
      - name: comment-spacings
        disabled: false
```

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04.2 LTS
 - go version go1.20.2 linux/amd64

**Comment by chavacava** [comment]
Hi @abemedia, thanks for filling the issue.

My understanding of the issue is you ask for `nolint` be part of the default accepted directives (alongside with `go` and `revive`). Is my understanding OK? If yes, feel free to make a PR.

**Comment by thaJeztah** [comment]
I don't think a hard-coded list should be used for these. Comments without a space are meant for "machine-readable" comments (such as `//nolint:`). Such comments are valid, but must follow the format described in https://github.com/golang/go/issues/37974#issuecomment-611088006

```
//[a-z0-9]+:[a-z0-9]
```

(`//` (comment), followed by a "namespace" (`[a-z0-9]+`), a colon (`:`), and some text (`[a-z0-9].` after it)

See;

- https://github.com/golang/go/issues/37974#issuecomment-611088006
- https://github.com/golang/go/issues/43776
- https://github.com/golang/go/issues/53835#issuecomment-1183507080

**Comment by chavacava** [comment]
Il will try to do a PR on this subject in the next days. My idea is to go ahead with the proposition by @thaJeztah (i.e. replacing  the hard coded list with a regex)

**Comment by ccoVeille** [comment]
Please note this change had a small consequence that some people might face too https://github.com/mgechev/revive/pull/986#issuecomment-2331719393
